src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -43,9 +51,7 @@ public void start(int periodicAggregateIntervalSec) {
         synchronized (emitMetricsLock) {
             // Cancel previously running task
             stop();
-            PutComponentMetricRequest request = new PutComponentMetricRequest();
-            request.setMetrics(metrics.collectMetrics());
-            future = ses.scheduleWithFixedDelay((Runnable) putComponentMetricOperationHandler.handleRequest(request),
+            future = ses.scheduleWithFixedDelay((Runnable) emitMetrics(),
```

Comment:
you can't just cast `emitMetrics()` to a runnable. What is this actually doing?

It is executing `emitMetrics()` immediately, then casting the response from `emitMetrics` which is `PutComponentMetricResponse` to a runnable, which will not succeed.

What you want is `this::emitMetrics`